guides/v2.3/javascript-dev-guide/javascript/requirejs.md

```diff
@@ -0,0 +1,105 @@
+
+## RequireJS in Magento {#requirejs-in-magento}
+
+### Configuration {#requirejs-config}
+
+All the configuration goes into the `requirejs-config.js` file. It has one root object `config` which can contain the configuration options described in below. All the configuration options are optional and are used when required only. The following snippet is a sample `requirejs-config.js` describing the structure of the file.
+
+```javascript
+var config = {
+    'map': {...},
+    'paths': {...}
+    'shim': {...},
+    'config': {
+        'mixins': {...}
+    }
+}
+```
+
+#### Map {#requirejs-config-map}
+
+The `map` configuration aliases any real AMD module that calls `define()` , to the specified alias for purpose of simplicity and ease of use. In the below snippet, `*` means all loaded RequireJS modules can use the specified alias.
+
+```javascript
+'map': {
+    '*': {
+        'alias': 'Vendor_Module/js/complex/path/amd-module'
+    }
+}
+```
+
+Now we can use our `Vendor_Module/js/complex/path/module` using `alias` in any RequireJS module or config file without needing to type the entire path.
+
+#### Paths {#requirejs-config-paths}
+
+The `paths`  configuration, similar to `map` configuration is used for aliasing not just any real AMD module that calls `define()`, but also any JS file (event from a URL), HTML templates, etc. Magento heavily uses this to alias URLs and also for third party libraries. Again, it is done only for the purpose of simplicity and ease of use.
+
+```javascript
+'paths': {
+    'alias': 'library/file',
+    'another-alias': 'https://some-library.com/file'
+}
+```
+
+#### Deps {#requirejs-config-deps}
+
+The `deps` configuration is used to add a dependency. It can either be used directly under `var config = {}` or under a [shim configuration](#requirejs-config-shim). Adding modules under an independent `deps` configuration will load the specified modules in all pages.
+
+In the below snippet, the custom `Vendor_Module/js/module` will be loaded in all pages.
+
+```javascript
+'deps': ['Vendor_Module/js/module']
+```
+
+#### Shim {#requirejs-config-shim}
+
+The `shim` configuration is used to build a dependency on a third party library, since we cannot modify it.
+
+When to use the `shim` configuration:
+
+-  To add a new dependency to a third party library
+-  To add a new dependency to a third party library which doesn't use an AMD module
+-  To change load order by adding a dependency to a third party library
+
+In the below snippet, dependencies can be added directly in an array, or it can be specified as an array under the `deps` key. The `exports` key is used to specify under what identifier the module is exported into. This export identifier can be used to access it.
+
+```javascript
+'shim': {
+    '3rd-party-library': ['myJSFile'],
```

Comment:
Hello @rogyar, I've made a better modification to that piece of code instead of adding a whole new snippet. I feel this is more efficient. Please let me know if it's alright. Thank you.

Reply:
That's much better, thank you